presto-main/src/main/java/com/facebook/presto/sql/planner/plan/ExchangeNode.java

```diff
@@ -62,14 +64,17 @@
     // for each source, the list of inputs corresponding to each output
     private final List<List<Symbol>> inputs;
 
+    private final Optional<OrderingScheme> orderingScheme;
+
```

Comment:
What we had before was a separate PlanNode for merging (MergeExchangeNode). However, ExchangeNode and MergeExchangeNode should behave pretty much the same for all of the optimizer rules, so you have a ton of duplication and opportunity for error if you introduce a new node.